message/messaging_service.cc

```diff
@@ -40,7 +40,6 @@
 #include "streaming/stream_mutation_fragments_cmd.hh"
 #include "cache_temperature.hh"
 #include "raft/raft.hh"
-#include "service/raft/messaging.hh"
```

Comment:
Re pinger:

It should be changed similarly to the rest of the code base: don't identify nodes using IPs anymore but using Raft IDs, and do the translation to IPs by accessing the address map when sending messages.

Reply:
We discussed this further on a call. There are two ways of thinking about this.

My original way of thinking is that the failure detector checks the liveness of **raft servers**. So the fd pings raft servers. In that case, the set of failure detector entries should be equal to the current set of raft servers, and we translate the raft ID to an IP using the address map at  the last moment possible, i.e. before we send a message inside the `ping` function.
To support IP address change while preserving this original way of thinking we should change the interface of `pinger` to operate on `raft::server_id`s instead of `gms::inet_address`es.

But there's a different way of thinking which @kostja prefers: the failure detector checks the liveness of IP addresses, not Raft servers. Thus, the FD pings IP addresses. The set of failure detector entries is then equal to some set of IP addresses, which should generally contain an IP address of every Raft server in the current group 0 configuration according to the current address map. When we want to check the liveness of a Raft server, we use the address map to translate it to IP and then check the liveness of IP.

I'm not sure which one is better.
@kostja mentioned that if we want `gossiper` to also use the failure detector to mark nodes up/down, the second way is better. However, I think if we really want to make the Raft ID a true (immutable) identifier for a node, the first way is fine; the gossiper would also identify nodes with Raft IDs so it would also be compatible with the first way.

Reply:
I prefer the first one. It is much easier to reason about the system this way.

Reply:
I believe I did it according to your preferences, the changes are on top, please check it out.